modules/history/src/views/full/message.jsx

```diff
@@ -0,0 +1,170 @@
+  return (
+    <div className={style['message-list']}>
+      <div className={style['no-conv']}>
+        <h3>No conversations selected</h3>
+        <p>
+          Please select a conversation on the left pane to see a message history. If there are no conversations
+          available, try talking to your bot and refresh conversations by clicking on the round arrow
+        </p>
+      </div>
+    </div>
+  )
+}
+
+export class MessagesViewer extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      inspectorIsShown: false,
+      currentlyFocusedMessage: null
+    }
+  }
+
+  getLastMessageDate = messageGroups => {
+    const messages = messageGroups.flatMap(m => m)
+    const maxDateMessage = _.maxBy(messages, m => m.createdOn)
+    return new Date(maxDateMessage.createdOn)
+  }
+
+  closeInspector = () => {
+    this.setState({ inspectorIsShown: false })
+  }
+
+  focusMessage(m) {
+    this.setState({ currentlyFocusedMessage: m, inspectorIsShown: true })
+  }
+
+  render() {
+    if (!this.props.convId || this.props.messageGroups.length <= 0) {
+      return <NoConversationSelected />
+    }
+    return (
+      <div className={style['message-viewer']}>
+        <div
+          className={classnames(
+            style['message-list'],
+            this.state.inspectorIsShown ? style['message-list-partial'] : style['message-list-full']
+          )}
+        >
+          <div className={style['message-header']}>
+            <div>
+              {this.props.convId && <div className={style['message-title']}>Conversation #{this.props.convId}</div>}
+              {this.props.convId && (
+                <div className={style['message-lastdate']}>
+                  Last message on : #{this.getLastMessageDate(this.props.messageGroups).toDateString()}
+                </div>
+              )}
+            </div>
+            <div className={style['message-header-icons']}>
+              <div className={style['message-header-icon_item']}>
+                <a
+                  href={this.props.fileURL}
+                  download="message_history"
+                  style={{
+                    color: '#233abc'
+                  }}
+                >
+                  <MdFileDownload />
+                </a>
+              </div>
+              <div className={style['message-header-icon_item']}>
+                <CopyToClipboard text={window.location.href}>
+                  <FiLink />
+                </CopyToClipboard>
+              </div>
+            </div>
+          </div>
+          {this.props.messageGroups &&
+            this.props.messageGroups.map(group => {
+              return <MessageGroup key={group[0].id} messages={group} focusMessage={this.focusMessage.bind(this)} />
+            })}
+        </div>
+        <div
+          className={classnames(
+            style['message-inspector'],
+            this.state.inspectorIsShown ? '' : style['message-inspector-hidden']
```

Comment:
with classnames, you can use boolean directly, eg:
``classnames(style.messageInspector, { style.messageInspectorHidden: this.state.isInspectorVisible }) ``